pkgs/development/libraries/libidn2/default.nix

```diff
@@ -20,9 +22,19 @@ stdenv.mkDerivation rec {
 
   patches = optional stdenv.isDarwin ./fix-error-darwin.patch;
 
+  # hack found here: https://github.com/sethrj/spack/commit/93393fe2f510a8ed9ace65534ee80e5433e19943
+  # works around the fact that autoreconf'ing libidn2 causes it to insist on gtk-doc being present
+  GTKDOCIZE = "echo";
+
   # The above patch causes the documentation to be regenerated, so the
   # documentation tools are required.
-  nativeBuildInputs = optionals stdenv.isDarwin [ help2man texinfo ];
+  nativeBuildInputs = [
+    # libidn2's Makefile forces an autoreconf if it notices changes to the ./configure script
```

Comment:
Where exactly is that and how does autoreconfHook solve this? I'm a bit confused by this and I wasn't able to find what you are referring to in the libidn2 tarball.

Reply:
Here is the error message you get without `autoreconfHook`; I'm still working on pinpointing the conditional that `libidn2` uses to decide that this is necessary:

```
make[3]: Entering directory '/build/libidn2-2.3.2/doc'
 cd .. && /nix/store/p4s4jf7aq6v6z9iazll1aiqwb34aqxq9-bootstrap-tools/bin/bash /build/libidn2-2.3.2/build-aux/missing automake-1.16 --gnu doc/Makefile
/build/libidn2-2.3.2/build-aux/missing: line 81: automake-1.16: command not found
WARNING: 'automake-1.16' is missing on your system.
         You should only need it if you modified 'Makefile.am' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'automake' program is part of the GNU Automake package:
         <https://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <https://www.gnu.org/software/autoconf>
         <https://www.gnu.org/software/m4/>
         <https://www.perl.org/>
make[3]: *** [Makefile:1414: Makefile.in] Error 127
make[3]: Leaving directory '/build/libidn2-2.3.2/doc'
make[2]: *** [Makefile:2122: Makefile.gdoc] Error 2
make[2]: Leaving directory '/build/libidn2-2.3.2/doc'
make[1]: *** [Makefile:1401: all-recursive] Error 1
make[1]: Leaving directory '/build/libidn2-2.3.2'
make: *** [Makefile:1309: all] Error 2
note: keeping build directory '/tmp/nix-build-libidn2-2.3.2.drv-1'
```

Reply:
It *is* the file mtimes.

This has led me to discover an even cleaner way to do this, which will be robust in the face of any other packages which do what `libidn2` does (although as I mentioned I have not found any).

Since I will be adding another commit to this PR, I'm going to "resolve" this conversation and continue below at the bottom of the discussion-chain so the other reviewers see the changes.

Thank you for prompting me to dig further into this!